app/controllers/auth_controller.rb

```diff
@@ -31,6 +31,15 @@ def fetch_return_to_path
   def do_user_auth(auth)
     sign_out
     user = User.from_oauth_hash(auth)
+    unless ENV["NO_WELCOME_EMAIL"]
```

Comment:
for all the tests. Every time a user gets created via FactoryGirl and then `login_as()` is called in a spec, a welcome email is getting generated, which was creating lots of test errors because `deliveries` was incrementing where not expected.